addon/components/nypr-accounts/basic-card.js

```diff
@@ -1,66 +1,64 @@
+import Ember from 'ember';
 import Component from 'ember-component';
-import { bool } from 'ember-computed';
 import layout from '../../templates/components/nypr-accounts/basic-card';
 import RSVP from 'rsvp';
 import Changeset from 'ember-changeset';
 import lookupValidator from 'ember-changeset-validations';
 import makeValidations from 'nypr-account-settings/validations/nypr-accounts/basic-card';
+import validateRemote from 'nypr-account-settings/validators/nypr-accounts/remote';
+import messages from 'nypr-account-settings/validations/nypr-accounts/custom-messages';
 import getOwner from 'ember-owner/get';
-import observer from 'ember-metal/observer';
-import { debounce } from 'ember-runloop';
 import get from 'ember-metal/get';
 import set from 'ember-metal/set';
 import computed from 'ember-computed';
+import { decamelize } from 'ember-string';
+import { task, timeout, all, waitForEvent } from 'ember-concurrency';
 
 export default Component.extend({
   layout,
   tagName: '',
-  isShowingModal: bool('resolveModal'),
   user: {},
-  verifyEmail: computed('user.email', 'changeset.email', function() {
-    return get(this, 'changeset.email') !== get(this, 'user.email');
-  }),
-
-  // until ember-changeset can handle debounce validations
-  // https://github.com/DockYard/ember-changeset/issues/102
-  // observe updates and debounce updating the changeset
-  // so that validations are run less frequently
-  // otherwise we get very unpleasant UI hiccups
-  usernameObserver: observer('preferredUsername', function() {
-    let newName = get(this, 'preferredUsername');
-    debounce(this, prefName => set(this, 'changeset.preferredUsername', prefName), newName, 150);
-  }),
+  debounceMs: Ember.testing ? 10 : 250,
 
   init() {
     this._super(...arguments);
-    let config = getOwner(this).resolveRegistration('config:environment');
-    let validations = makeValidations({usernamePath: config.wnycAuthAPI});
+    this.config = getOwner(this).resolveRegistration('config:environment');
+    let validations = makeValidations({validationPath: this.config.wnycAuthAPI});
 
     let user = get(this, 'user');
     let changeset = new Changeset(user, lookupValidator(validations), validations);
     this.changeset = changeset;
-
-    // provide a temporary binding for preferred username on the template
-    // so we can do remote async validations before setting to the
-    // changeset
-    this.preferredUsername = get(user, 'preferredUsername');
   },
 
+  emailWasChanged: computed('changeset.email', 'user.email', function() {
+    return get(this, 'changeset.email') !== get(this, 'user.email');
+  }),
+
+  usernameWasChanged: computed('changeset.preferredUsername', 'user.preferredUsername', function() {
+    return get(this, 'changeset.preferredUsername') !== get(this, 'user.preferredUsername');
+  }),
+
   onEmailChange() {
-    if (this.changeset.get('email')) {
+    let newEmail = this.changeset.get('email');
+    if (newEmail !== undefined) {
```

Comment:
from what I understand, this conditional is checking to see if the email value has changed from what it originally was. The else case discards changes to the email and confirmEmail fields. This gets around an issue where a changeset doesn't recognize if its value is the same as the original value of the object it's based upon.

There are two purposes here:
1. force confirmEmail to validate when the email changes
2. discard email a change so extraneous error messages do not appear.

one suggestion is to look at `emailWasChanged` instead of `changeset.email` for this conditional
another is to explain a bit more about what's going on, or else refactor to make it clearer.